extensions/kafka/src/main/java/io/deephaven/kafka/KafkaPublishOptions.java

```diff
@@ -108,25 +110,33 @@ public boolean publishInitial() {
 
     /**
      * The topic column. When set, uses the the given {@link CharSequence}-compatible column from {@link #table()} as
-     * the first source for setting the kafka record topic.
+     * the first source for setting the kafka record topic. When not present, or the column value is null,
+     * {@link #topic()} will be used.
      *
      * @return the topic column name
      */
     public abstract Optional<ColumnName> topicColumn();
 
     /**
      * The partition column. When set, uses the the given {@code int} column from {@link #table()} as the first source
-     * for setting the kafka record partition.
+     * for setting the kafka record partition. When not present, or the column value is null, {@link #partition()} will
+     * be used if present. If a valid partition number is specified, that partition will be used when sending the
+     * record. If no partition is specified but a key is present, a partition will be chosen using a hash of the key. If
```

Comment:
OK. Maybe we should clarify that the last two bits are "Kafka's default behavior" that we don't control?